### ISSUE TITLE
Issue with pet "chase" mechanic

### ISSUE THREAD

**Issue by Klimpb** [ccshiro/cc-buglist#548]
Tested on a mob but it seemed to be the same way vs players.

On Corecraft: http://webmup.com/db7d6/
On Retail TBC: http://webmup.com/f12c4/ (might be slow to load, alt link: http://bit.ly/1Fh4OH9)

On Corecraft the pets easily fall behind when the target he attacks moves which makes it so he's out of range and can no longer attack, sometimes he even stops for a second just to make his auto-attack.

but if you look at the retail video you can see that the pet is keeping up with the mage at all times and does not wait until he is out of range to follow the mage.

My guess on how it could have worked on retail mechanic wise: when target moves, pet starts to move.
Currently on Corecraft it seems like: when pet is out of range, pet starts to move.

**Comment by Chulain** [comment]
i'll do the rest later/tomorrow

**Comment by Chulain** [comment]
nice find

**Comment by DaRealBoss** [comment]
seems to be a common issue on other private servers as well

**Comment by ccshiro** [comment]
I intend to fix this prior to release, but I will leave it as is for next test. Need some time to consider how I best go about fixing this.

**Comment by Klimpb** [comment]
More proof: https://www.youtube.com/watch?v=dc0WrORiHmM
2:38 - 8:45

**Comment by ccshiro** [comment]
Also check out #120.

**Comment by Klimpb** [comment]
More videos of it:
https://youtu.be/R9UHt5t2orA

**Comment by Klimpb** [comment]
Tested it out a little bit today, it works much much better than before and it looks very much like it does in the retail videos. Though pet still falls behind a few times, not often but it happens some times when players change to the opposite direction W/A.

Not sure how reliable pets should be able to chase targets but I guess it could be tweaked at some point later on if necessary.

**Comment by Klimpb** [comment]
https://www.youtube.com/watch?v=dc0WrORiHmM
8:07 - 8:48

and this video again: https://www.youtube.com/watch?v=R9UHt5t2orA

Compare that to Corecraft: https://www.youtube.com/watch?v=nZDetZOfYX4

Sure I was sidestepping a bit and trying to get rid of the pet but I managed to get rid of the pet pretty much right away when I started. Doubt that is intended.

The pets in the retail videos acted more quickly and responded faster to movements and change of direction. At least that's what I can see.

I'm not sure if pet movement and mob movement were be the same.
Again, great job with the mechanic so far. It just needs some tweaking.
